core/src/main/java/org/jdbi/v3/core/MappingRegistry.java

```diff
@@ -87,10 +88,25 @@ public void addColumnMapper(ColumnMapperFactory factory) {
     }
 
     public Optional<ColumnMapper<?>> findColumnMapperFor(Type type, StatementContext ctx) {
-        return Optional.ofNullable(columnCache.computeIfAbsent(type, t ->
-                columnFactories.stream()
-                        .flatMap(factory -> toStream(factory.build(t, ctx)))
-                        .findFirst()
-                        .orElse(null)));
+        // ConcurrentHashMap can enter an infinite loop on nested computeIfAbsent calls.
+        // Since column mappers can wrap other column mappers, we have to populate the cache the old fashioned way.
+        // See https://bugs.openjdk.java.net/browse/JDK-8062841, https://bugs.openjdk.java.net/browse/JDK-8142175
+        ColumnMapper<?> mapper = columnCache.get(type);
+
+        if (mapper != null) {
+            return Optional.of(mapper);
+        }
+
+        mapper = columnFactories.stream()
+                .flatMap(factory -> toStream(factory.build(type, ctx)))
+                .findFirst()
+                .orElse(null);
+
+        if (mapper != null) {
+            columnCache.put(type, mapper);
+            return Optional.of(mapper);
```

Comment:
you could potentially avoid unboxing the `Optional` from the `findFirst` and reusing it